### ISSUE TITLE
2021 version, GPU core and clock offset doesn’t seem to work

### ISSUE THREAD

**Issue by TomBonnot** [seerge/g-helper#2439]
### Rules

- [X] I made myself familiar with the <a href='https://github.com/seerge/g-helper?tab=readme-ov-file'>Readme</a>, <a href='https://github.com/seerge/g-helper/wiki/FAQ'>FAQ</a> and <a href='https://github.com/seerge/g-helper/wiki/Troubleshooting'>Troubleshooting</a>.
- [X] I understand that, if insufficient information or no app logs will be provided, my issue will be closed without an answer.

### What's wrong?

I have been trying to push my dGPU a bit more for a long time with GHelper, I’ve never been checking that much, as far as I know passing over 160MHz core clock the game would crash and the dGPU restarting. 

I am trying to get better performance on Cyberpunk, so far my dGPU wouldn’t go pass 24W, meanwhile my CPU being capped at 30W on GHelper. Plus, the core and memory clock of my dGPU are not affected by GHelper since it is 2100MHz and 1600MHz, whatever offset I am trying to add. 

So I went to the log GHelper creates, and it seems to return an error code after applying core clock : 
15-Apr-24 11:32:22 AM: GET GPU CLOCKS: 160, 250
15-Apr-24 11:32:22 AM: GET CLOCK LIMIT: 0
15-Apr-24 11:32:22 AM: ReadGPUPowerBase: -65536
15-Apr-24 11:32:26 AM: GET CLOCK LIMIT: 0
15-Apr-24 11:32:26 AM: GET GPU CLOCKS: 160, 250
15-Apr-24 11:32:26 AM: SET GPU CLOCKS: 155, 250
15-Apr-24 11:32:26 AM: GET GPU CLOCKS: 155, 250
15-Apr-24 11:32:26 AM: GET CLOCK LIMIT: 0
15-Apr-24 11:32:26 AM: ReadGPUPowerBase: -65536
15-Apr-24 11:35:56 AM: GET GPU CLOCKS: 155, 250
15-Apr-24 11:35:56 AM: GET CLOCK LIMIT: 0
15-Apr-24 11:35:56 AM: ReadGPUPowerBase: -65536
15-Apr-24 11:36:00 AM: PowerLimit A3 = 20 : OK
15-Apr-24 11:36:00 AM: PowerLimit A0 = 20 : OK

Tho, the Get is reading okay, but the ReadGPUPowerBase is surely returning an error. 

### How to reproduce the bug?

1. Go To Fans + Power 
2. Go To GPU 
3. Use the slider to add more frequency to the dGPU
4. Open Ghelper log.txt

### Logs

[log.txt](https://github.com/seerge/g-helper/files/14981283/log.txt)

### Device and Model

G14 GA401QM

### Armoury Crate

Uninstalled

### Asus Services

None

### Version

0.166.0

### OS

Windows 11 22H2

**Comment by seerge** [comment]
@TomBonnot hello, 

I see that app sets core / memory clock offsets and they work - cause after setting it reads them from the driver, and driver returns correct values

```
15-Apr-24 11:32:26 AM: GET CLOCK LIMIT: 0  <---- NO CLOCK LIMIT (i.e. everything is ok)
15-Apr-24 11:32:26 AM: GET GPU CLOCKS: 160, 250 <--- previous values
15-Apr-24 11:32:26 AM: SET GPU CLOCKS: 155, 250
15-Apr-24 11:32:26 AM: GET GPU CLOCKS: 155, 250 <---- IT READS NEW VALUES, THEY ARE CORRECT
```

G-Helper doesn't limit your GPU power anyhow (and it can't). 

Where do you test all this ? It could be that game that you use is just CPU-limited (and your limits are quite low) 

Try to run GPU-only benchmark like Furmark or so and monitor consumption there ?

**Comment by seerge** [comment]
@TomBonnot 

> That’s what I understand from the log, tho I don’t know what the -65536 means.

That's app checks if there is a an option to set variable GPU power (only applicable for 2024 models). It has no effect on you )) and value means that that option is not present. So just ignore it.

As for clock offsets - what it does, it sets a setting in Nvidia driver (via it's official API) - so for old power usage, gpu will go a bit higher in it's clock.

It means that you get a bit more performance for **same power** . But if GPU doesn't need max out clock - you won't see it (but then it will consume **less power** than it would otherwise w/o offset)

---

So in short - everything works as it should, app sets values and they are reflected in a driver :) rest is up to actual app / game on how to utilize that :)

Thanks

**Comment by TomBonnot** [comment]
Alright thank you. 
I wasn't sure it was working or no, better be safe than sorry, thank you for all the work. 

I have some things to do because it is not normal that giving negative offset numbers for the memory clock makes the dGPU crash, and giving extra, makes the dGPU lagging ! Maybe it's my device ! 

Thanks anyway !

**Comment by seerge** [comment]
@TomBonnot 

Memory clock offset is usually reflected in monitoring tools immediately (as memory mostly run on same clock all the time).

It's strange that it's lagging, but it could be that GPU is just not designed to run on different clocks ...

You can try to do a [clean driver reinstall with DDU ](https://github.com/seerge/g-helper/wiki/Troubleshooting#reinstalling-gpu-drivers) , to exclude any possible driver hiccups :)
